### PR TITLE
Test: Multi Draft Upload

### PR DESCRIPTION
**Pull request by bigvaimarcella**
# Description

## Checklist

- [ x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ x] I have checked that unit tests pass locally with my changes
- [x ] I have run the npm script lint on the frontend and backend
- [x ] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)
- [x ] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+      try {
+        let reader = new FileReader();
+        reader.onload = function (e) {
+          this.Json = JSON.parse(e.target.result);
+        }.bind(this);
+
+        reader.onloadend = () => {
+          this.preValidateSubmission();
+        };
+        reader.readAsText(this.file);
```
Comment (jasonchung1871) [inline app/frontend/src/components/designer/FormViewerMultiUpload.vue:266]: this line should come after prevalidateSubmission. onloadend is a promise, so it'll execute preValidateSubmission but can also execute this line at the same time before preValidateSubmission is complete.

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+              });
+            }
+          } catch (error) {
+            errors.push({
+              submission: this.index,
+              message: this.ERROR.ERROR_WHILE_CHECKVALIDITY,
+            });
+          }
+          this.index++;
+          this.value = this.pourcentage(this.index);
```
Comment (jasonchung1871) [inline app/frontend/src/components/designer/FormViewerMultiUpload.vue:334]: please try to keep the language consistent across the application. this should be `percentage`.

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+          if (this.index < this.Json.length) {
+            this.validate(this.Json[this.index], errors);
+          } else {
+            this.endValidation(errors);
+          }
+        }.bind(this),
+        12
+      );
+    },
+    pourcentage(i) {
```
Comment (jasonchung1871) [inline app/frontend/src/components/designer/FormViewerMultiUpload.vue:345]: please try to keep the language consistent across the application. this should be `percentage`.

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue (hunk 1 above)
Comment (bigvaimarcella) [reply app/frontend/src/components/designer/FormViewerMultiUpload.vue:266]: this line is entrypoint  imperatively  it gonna run before onloadend cause onloadend is a callback its call after finishing to read the file .

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue (hunk 2 above)
Comment (bigvaimarcella) [reply app/frontend/src/components/designer/FormViewerMultiUpload.vue:334]: Solved

File: app/frontend/src/components/designer/FormViewerMultiUpload.vue (hunk 3 above)
Comment (bigvaimarcella) [reply app/frontend/src/components/designer/FormViewerMultiUpload.vue:345]: Solved

File: app/frontend/src/components/designer/FormViewer.vue [hunk 4]
```diff
@@ -156,10 +223,30 @@ export default {
...
       versionIdToSubmitTo: this.versionId,
+      allowSubmitterToUploadFile: false,
+      formFields: [],
+      json_csv: {
+        data: [],
+        file_name: String,
+      },
+      bulkFile: false,
+      formElement: undefined,
+      sbdMessage: {
```
[the remark this answers is not in the document]
Comment (bigvaimarcella) [reply app/frontend/src/components/designer/FormViewer.vue:234]: Solved i add a comment

File: app/frontend/package.json [hunk 5]
```diff
@@ -35,6 +35,7 @@
     "bootstrap-scss": "^4.6.2",
     "core-js": "^3.30.2",
     "deep-freeze": "^0.0.1",
+    "export-from-json": "^1.7.2",
```
[the remark this answers is not in the document]
Comment (bigvaimarcella) [reply app/frontend/package.json:38]: Solved

**Pull request merged**